WooCommerce/Classes/ViewRelated/BottomSheet/ListSelector/BottomSheetListSelectorViewController.swift

```diff
@@ -0,0 +1,127 @@
+import UIKit
+
+/// Displays a list (implemented by table view) for the user to select a generic model from a bottom sheet.
+///
+final class BottomSheetListSelectorViewController<DataSource: BottomSheetListSelectorDataSource, Model, Cell>:
+UIViewController, UITableViewDataSource, UITableViewDelegate where DataSource.Model == Model, DataSource.Cell == Cell {
+    private let viewProperties: BottomSheetListSelectorViewProperties
+    private var dataSource: DataSource
+    private let onDismiss: (_ selected: Model?) -> Void
+
+    private let rowType = Cell.self
+
+    private let estimatedSectionHeight = CGFloat(44)
+
+    @IBOutlet private(set) weak var tableView: UITableView!
+
+    init(viewProperties: BottomSheetListSelectorViewProperties,
+         dataSource: DataSource,
+         onDismiss: @escaping (_ selected: Model?) -> Void) {
+        self.viewProperties = viewProperties
+        self.dataSource = dataSource
+        self.onDismiss = onDismiss
+        super.init(nibName: "BottomSheetListSelectorViewController", bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureMainView()
+        configureTableView()
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        onDismiss(dataSource.selected)
+        super.viewWillDisappear(animated)
+    }
+
+    // MARK: UITableViewDataSource
+    //
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return dataSource.data.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: rowType.reuseIdentifier,
+                                                       for: indexPath) as? Cell else {
+                                                        fatalError()
+        }
+        let model = dataSource.data[indexPath.row]
+        dataSource.configureCell(cell: cell, model: model)
+
+        return cell
+    }
+
+    // MARK: UITableViewDelegate
+    //
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+
+        let selected = dataSource.data[indexPath.row]
+        if selected != dataSource.selected {
```

Comment:
great question! this `isSelected` function is from the pre-existing `ListSelectorDataSource`/`PaginatedListSelectorDataSource`, and because an implementation's model could have a different `Equatable` implementation from what's being considered as equal in the list selector UI:

https://github.com/woocommerce/woocommerce-ios/blob/548372e3947701ceb6d013c0eafce445223b6877/WooCommerce/Classes/ViewRelated/Products/Edit%20Product/List%20Selector%20Data%20Source/ProductTaxClassListSelectorDataSource.swift#L31-L33

vs. `TaxClass`'s equatable implementation:

https://github.com/woocommerce/woocommerce-ios/blob/548372e3947701ceb6d013c0eafce445223b6877/Networking/Networking/Model/TaxClass.swift#L46-L52

however, I think it's very unlikely the tax class's name/slug would be outdated since we also sync the tax classes before rendering the list.

for this bottom sheet list selector UI, let's start simple with your suggestion and we can talk about options when special circumstances come up  👍

Reply:
Sounds good! I have some ideas about how we can work around that and still use `Equatable`. But, yeah, let's discuss it when it actually happens. 😄